src/utils/return_periods.py

```diff
@@ -1,8 +1,111 @@
 import numpy as np
 import pandas as pd
 from lmoments3 import distr, lmom_ratios
+from scipy.interpolate import interp1d
 from scipy.stats import norm, pearson3, skew
 
+# Empirical RP Functions
+
+
+def fs_add_rp(df, df_maxima, by):
+    df_nans, df_maxima_nans = [
+        extract_nan_strata(x, by=by) for x in [df, df_maxima]
+    ]
+
+    df_filt = df[
+        ~df[by].apply(tuple, axis=1).isin(df_nans.apply(tuple, axis=1))
+    ]
+    df_maxima_filt = df_maxima[
+        ~df_maxima[by]
+        .apply(tuple, axis=1)
+        .isin(df_maxima_nans.apply(tuple, axis=1))
+    ]
```

Comment:
I found the source of the `SettingWithCopyWarning` 👀 

To avoid those coming up, can just add `.copy()` to each of these filtering steps like:

```
    df_filt = df[
        ~df[by].apply(tuple, axis=1).isin(df_nans.apply(tuple, axis=1))
    ].copy()
    df_maxima_filt = df_maxima[
        ~df_maxima[by]
        .apply(tuple, axis=1)
        .isin(df_maxima_nans.apply(tuple, axis=1))
    ].copy()
```

Anyways not necessary, up to you if you want to switch it.

Reply:
yea i had gotten rid of a bunch of them by using `.loc` , but not sure i how i could do that for these operations. Anyways i added the `.copy()` for now